### PR TITLE
JAMES-2578 Fix warning using Mailet new API

### PR DESCRIPTION
**Pull request by blackheaven**

### CODE REVIEW

**Comment by trantienduchn** [comment]
```
[5125170d793ebe03f04d8ea7745b0386ad64b3f6] [ERROR] Errors: 
[5125170d793ebe03f04d8ea7745b0386ad64b3f6] [ERROR]   WithStorageDirectiveTest.serviceShouldAddDeliveryPathForRecipients:67 » IllegalState
[5125170d793ebe03f04d8ea7745b0386ad64b3f6] [ERROR]   WithStorageDirectiveTest.serviceShouldNotThrowWhenNoRecipients:88 » IllegalState
[5125170d793ebe03f04d8ea7745b0386ad64b3f6] [ERROR]   WithStorageDirectiveTest.serviceShouldOverridePreviousStorageDirectives:109 » IllegalState
[5125170d793ebe03f04d8ea7745b0386ad64b3f6] [INFO] 
[5125170d793ebe03f04d8ea7745b0386ad64b3f6] [ERROR] Tests run: 811, Failures: 0, Errors: 3, Skipped: 0
```

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/WithStorageDirective.java [hunk 1]
```diff
@@ -61,12 +64,14 @@ public WithStorageDirective(UsersRepository usersRepository) {
 
     @Override
     public void init() throws MessagingException {
-        targetFolderName = getInitParameter(TARGET_FOLDER_NAME);
-        validateMailetConfiguration();
+        targetFolderName = AttributeValue.of(validateMailetConfiguration(TARGET_FOLDER_NAME));
+        validateMailetConfiguration(getInitParameter(TARGET_FOLDER_NAME));
```
Comment (Arsnael) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/WithStorageDirective.java]: You do a `getInitParamater` as well in `validateMailetConfiguration`. No need of doing it twice right?

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/WithStorageDirective.java [hunk 2]
```diff
@@ -61,12 +64,14 @@ public WithStorageDirective(UsersRepository usersRepository) {
...
     }
 
-    public void validateMailetConfiguration() {
-        Preconditions.checkState(!Strings.isNullOrEmpty(targetFolderName), "You need to specify " + TARGET_FOLDER_NAME);
+    public String validateMailetConfiguration(String initParameterName) {
```
Comment (Arsnael) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/WithStorageDirective.java]: This method doesn't seem to be used outside of this class, shouldn't be public

**Comment by rouazana** [comment]
rebased to solve conflicts

**Comment by chibenwa** [comment]
Héhé you need to rebase again

**Comment by rouazana** [comment]
> Héhé you need to rebase again

Done :)

**Comment by Arsnael** [comment]
```
[3ee13cf27faa68714bb6fd79f7eda5dbb2d20bf6] [ERROR] Failures: 
[3ee13cf27faa68714bb6fd79f7eda5dbb2d20bf6] [ERROR]   ManageableMailQueueContract.browseShouldReturnMailsWithAttributes:472 
[3ee13cf27faa68714bb6fd79f7eda5dbb2d20bf6] The following 2 assertions failed:
[3ee13cf27faa68714bb6fd79f7eda5dbb2d20bf6] 1) 
[3ee13cf27faa68714bb6fd79f7eda5dbb2d20bf6] Expecting:
[3ee13cf27faa68714bb6fd79f7eda5dbb2d20bf6]   <Optional[Attribute{name=AttributeName{name=Attribute Name 1}, value=AttributeValue{value=Attribute Value 1, serializer=FSTSerializer}}]>
[3ee13cf27faa68714bb6fd79f7eda5dbb2d20bf6] to contain:
[3ee13cf27faa68714bb6fd79f7eda5dbb2d20bf6]   <Attribute{name=AttributeName{name=Attribute Name 1}, value=AttributeValue{value=Attribute Value 1, serializer=StringSerializer}}>
[3ee13cf27faa68714bb6fd79f7eda5dbb2d20bf6] but did not.
[3ee13cf27faa68714bb6fd79f7eda5dbb2d20bf6] at ManageableMailQueueContract.lambda$browseShouldReturnMailsWithAttributes$15(ManageableMailQueueContract.java:474)
[3ee13cf27faa68714bb6fd79f7eda5dbb2d20bf6] 2) 
[3ee13cf27faa68714bb6fd79f7eda5dbb2d20bf6] Expecting:
[3ee13cf27faa68714bb6fd79f7eda5dbb2d20bf6]   <Optional[Attribute{name=AttributeName{name=Attribute Name 2}, value=AttributeValue{value=Attribute Value 2, serializer=FSTSerializer}}]>
[3ee13cf27faa68714bb6fd79f7eda5dbb2d20bf6] to contain:
[3ee13cf27faa68714bb6fd79f7eda5dbb2d20bf6]   <Attribute{name=AttributeName{name=Attribute Name 2}, value=AttributeValue{value=Attribute Value 2, serializer=StringSerializer}}>
[3ee13cf27faa68714bb6fd79f7eda5dbb2d20bf6] but did not.
[3ee13cf27faa68714bb6fd79f7eda5dbb2d20bf6] at ManageableMailQueueContract.lambda$browseShouldReturnMailsWithAttributes$15(ManageableMailQueueContract.java:476)
[3ee13cf27faa68714bb6fd79f7eda5dbb2d20bf6] 
[3ee13cf27faa68714bb6fd79f7eda5dbb2d20bf6] [ERROR]   MailQueueContract.queueShouldPreserveMailAttribute:161 
[3ee13cf27faa68714bb6fd79f7eda5dbb2d20bf6] Expecting:
[3ee13cf27faa68714bb6fd79f7eda5dbb2d20bf6]   <Optional[Attribute{name=AttributeName{name=any}, value=AttributeValue{value=value, serializer=FSTSerializer}}]>
[3ee13cf27faa68714bb6fd79f7eda5dbb2d20bf6] to contain:
[3ee13cf27faa68714bb6fd79f7eda5dbb2d20bf6]   <Attribute{name=AttributeName{name=any}, value=AttributeValue{value=value, serializer=StringSerializer}}>
[3ee13cf27faa68714bb6fd79f7eda5dbb2d20bf6] but did not.
```
test this please

**Comment by rouazana** [comment]
merged

**Pull request closed without merging**